server/controller/admin/order/orderController.js

```diff
@@ -0,0 +1,272 @@
+const User = require("../../../model/userModel");
+const Order = require("../../../model/orderModel");
+
+const createOrder = async (req, res) => {
+  console.log(req.userId);
+  console.log(req.email);
+  const {
+    item,
+    size,
+    quantity,
+    price,
+    totalAmount,
+    depositAmount,
+    deliveredItems,
+    //get vendor ID but display name in frontend
+    vendor,
+    substituteVendor,
+    needed,
+    status,
+    purchaseOrdered,
+  } = req.body;
+
+  try {
+    const existingUser = await User.findOne({ _id: req.userId });
+
+    // console.log(existingUser);
+    if (!existingUser) {
+      return res.status(404).json({ error: "No User found" });
+    }
+    if (
+      !item ||
+      !size ||
+      !quantity ||
+      !price ||
+      !totalAmount ||
+      !depositAmount ||
+      !deliveredItems ||
+      !vendor ||
+      !needed ||
+      !status ||
+      !purchaseOrdered
+    ) {
+      return res.status(400).json({
+        error: "All required fields must be provided for creating Order!",
+      });
+    }
+
+    const newOrderData = {
+      item: item,
+      size: size,
+      quantity: quantity,
+      price: price,
+      totalAmount: totalAmount,
+      depositAmount: depositAmount,
+      deliveredItems: deliveredItems,
+      //will use vendor ID after creating real vendor (extracting vendor ID from Vendor schema in DB)
+      vendor: req.userId,
+      date: Date.now(),
+      needed: needed,
+      status: true,
+      approved: false,
+      createdBy: req.userId,
+      purchaseOrdered: purchaseOrdered,
+      ordered: true,
```

Comment:
```suggestion
      ordered: false,
```
ordered initial value should also be set to false